src/context/client.rs

```diff
@@ -19,48 +20,92 @@ impl RedisModuleClientInfo {
 }
 
 /// GetClientNameById, GetClientUserNameById and GetClientCertificate use autoMemoryAdd on the ValkeyModuleString pointer
-// after the callback (command, server event handler, ...) these ValkeyModuleString pointers will be freed automatically
+/// after the callback (command, server event handler, ...) these ValkeyModuleString pointers will be freed automatically
 impl Context {
     pub fn get_client_id(&self) -> u64 {
         unsafe { RedisModule_GetClientId.unwrap()(self.ctx) }
     }
 
-    pub fn get_client_name(&self) -> ValkeyString {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientNameById
+    pub fn get_client_name_by_id(&self, client_id: u64) -> ValkeyString {
         let client_name = unsafe { RedisModule_GetClientNameById.unwrap()(self.ctx, client_id) };
         ValkeyString::from_redis_module_string(self.ctx, client_name)
     }
 
-    pub fn set_client_name(&self, client_name: &ValkeyString) -> i64 {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientNameById using current client ID
+    pub fn get_client_name(&self) -> ValkeyString {
+        self.get_client_name_by_id(self.get_client_id())
+    }
+
+    /// wrapper for RedisModule_SetClientNameById
+    pub fn set_client_name_by_id(&self, client_id: u64, client_name: &ValkeyString) -> i64 {
         let resp = unsafe { RedisModule_SetClientNameById.unwrap()(client_id, client_name.inner) };
         resp as i64
     }
 
-    pub fn get_client_username(&self) -> ValkeyString {
-        let client_id = self.get_client_id();
-        let client_username =
-            unsafe { RedisModule_GetClientUserNameById.unwrap()(self.ctx, client_id) };
-        ValkeyString::from_redis_module_string(self.ctx, client_username)
+    /// wrapper for RedisModule_SetClientNameById using current client ID
+    pub fn set_client_name(&self, client_name: &ValkeyString) -> i64 {
+        self.set_client_name_by_id(self.get_client_id(), client_name)
     }
 
+    /// wrapper for RedisModule_GetClientUserNameById
     pub fn get_client_username_by_id(&self, client_id: u64) -> ValkeyString {
         let client_username =
             unsafe { RedisModule_GetClientUserNameById.unwrap()(self.ctx, client_id) };
```

Comment:
Same comment as other places - We can return a result here to handle the cases where the API returns an error

Reply:
agree, updated the code